dash_bootstrap_components/table.py

```diff
@@ -0,0 +1,86 @@
+from collections import Sized
+
+import dash_html_components as html
+
+
+def _generate_table_from_df(
+    cls,
+    df,
+    float_format=None,
+    columns=None,
+    header=True,
+    index=False,
+    index_label=None,
+    date_format=None,
+    **table_kwargs
+):
+    """
+    Generate a Table component from a dataframe.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        DataFrame to render as a table.
+    float_format : str, optional
+        Format to use for floating point numbers.
+    columns : sequence, optional
+        Columns to render.
+    header : boolean or list(str) or dict(str: str), optional
+        Write out the column names. If a list of strings is given it is assumed
+        to be aliases for the columns names (and hence must be the same length
+        as df.columns). A dict can be passed to rename some columns, the format
+        is {'<current_name>': '<new_name>'}. The dictionary need not have an
+        entry for every column.
+    index : boolean, optional
+        Render the row names (index).
+    index_label : str, optional
+        Column label for index column if desired. If None is passed, but both
+        header and index are True, then the index name is used.
+    date_format : str, optional
+        Format string for datetime objects.
+    **table_kwargs : Keyword arguments
+        Additional arguments to pass to the table component. See
+        dash_bootstrap_components.Table for details.
+    """
+    if columns is not None:
+        df = df.loc[:, columns]
+
+    if float_format is not None:
+        for c in df.select_dtypes(["float"]).columns:
+            df[c] = df[c].map(lambda x: "{1:{0}}".format(float_format, x))
+
+    if date_format is not None:
+        for c in df.select_dtypes(["datetime"]).columns:
+            df[c] = df[c].map(lambda x: x.strftime(date_format))
+
+    if index:
+        df = df.reset_index()
+        if index_label is not None:
+            df = df.rename(columns={"index": index_label})
+
+    if header:
+        if isinstance(header, dict):
+            df = df.rename(columns=header)
+        elif isinstance(header, Sized):
```

Comment:
@pbugnion I previously was checking if header was an instance of `(tuple, list, np.ndarray, pd.Index)`, but since we don't want `pandas` or `numpy` to be dependencies I need an alternative that can be used without importing either. This was the best I came up with, is there a better way do you think?

Reply:
To clarify, I think it's fine to have pandas and numpy as dependencies for this part of the code. We could just nest the import inside the function so it's only imported when the function gets used, not when the module gets loaded.

My only concern was with explicitly specifying pandas as dependencies in `install_requires` in `setup.py`. That would force users to install pandas, whether they use `_generate_table_from_df` or not. I believe the most common way to specify optional dependencies is with `extras_require` (see [setuptools docs](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies)).

Reply:
Ah right, I didn't think to put the imports inside the function, and was worried about pandas getting imported whenever you imported the module, that makes a lot more sense. Will amend!